camel/storages/vectordb_storages/milvus.py

```diff
@@ -0,0 +1,360 @@
+        """
+        if self._collection_exists(self.collection_name):
+            in_dim = self._get_collection_info(
+                self.collection_name)["vector_dim"]
+            if in_dim != self.vector_dim:
+                # The name of collection has to be confirmed by the user
+                raise ValueError(
+                    "Vector dimension of the existing collection "
+                    f'"{self.collection_name}" ({in_dim}) is different from '
+                    f"the given embedding dim ({self.vector_dim}).")
+        else:
+            self._create_collection(collection_name=self.collection_name, )
+
+    def _create_collection(
+        self,
+        collection_name: str,
+        **kwargs: Any,
+    ) -> None:
+        r"""Creates a new collection in the database.
+
+        Args:
+            collection_name (str): Name of the collection to be created.
+            **kwargs (Any): Additional keyword arguments.
+        """
+        from pymilvus import CollectionSchema, DataType, FieldSchema
+
+        index_params = {
+            "metric_type": "COSINE",
+            "index_type": "AUTOINDEX",
+            "params": {},
+        }
+
+        fields = [
+            FieldSchema(name='id', dtype=DataType.VARCHAR,
+                        descrition='A unique identifier for the vector',
+                        is_primary=True, auto_id=False, max_length=65535),
+            FieldSchema(
+                name='payload', dtype=DataType.VARCHAR,
+                description=('Any additional metadata or information related'
+                             'to the vector'), max_length=65535),
+            FieldSchema(
+                name='vector', dtype=DataType.FLOAT_VECTOR,
+                description='The numerical representation of the vector',
+                dim=self.vector_dim)
+        ]
+        schema = CollectionSchema(fields=fields,
+                                  description='Collection schema')
+
+        self._client.create_collection_with_schema(
+            collection_name=collection_name,
+            schema=schema,
+            index_params=index_params,
+            **kwargs,
+        )
+
+    def _delete_collection(
+        self,
+        collection_name: str,
+    ) -> None:
+        r"""Deletes an existing collection from the database.
+
+        Args:
+            collection (str): Name of the collection to be deleted.
+        """
+        self._client.drop_collection(collection_name=collection_name)
+
+    def _collection_exists(self, collection_name: str) -> bool:
+        r"""Checks whether a collection with the specified name exists in the
+        database.
+
+        Args:
+            collection_name (str): The name of the collection to check.
+
+        Returns:
+            bool: True if the collection exists, False otherwise.
+        """
+        for c in self._client.list_collections():
+            if collection_name == c:
+                return True
+        return False
```

Comment:
Better use `has_collection`: https://milvus.io/api-reference/pymilvus/v2.3.x/MilvusClient/Collections/has_collection.md

Reply:
I'm using pymilvus 2.3.6, 'MilvusClient' object has no attribute 'has_collection' as documentation showed

Reply:
We recommend to use MilvusClient with pymilvus>=2.3.7.

Reply:
has update to 2.3.7 using `has_collection`